src/wp-admin/includes/image.php

```diff
@@ -1143,3 +1223,31 @@ function _copy_image_file( $attachment_id ) {
 
 	return $dst_file;
 }
+
+/**
+ * Returns an array with the list of valid mime types that a specific mime type should be converted into.
+ * For example an `image/jpeg` should be converted into an `image/jpeg` and `image/webp`. The first type
+ * is considered the primary output type for this image.
+ *
+ * @since 6.0.0
+ *
+ * @param $attachment_id int The attachment ID.
+ * @return array<string, array<string>> An array of valid mime types, where the key is the mime type and the value is the extension type.
+ */
+function wp_upload_image_mime_transforms( $attachment_id ) {
+	$image_mime_transforms = array(
+		'image/jpeg'      => array( 'image/jpeg', 'image/webp' ),
+		'image/webp'      => array( 'image/webp', 'image/jpeg' ),
+		'application/pdf' => array( 'image/jpeg' ),
```

Comment:
right, without this pdfs were failing to be converted to jpegs as they are currently (when Imagick is present). Ill test again without this should be resolved now.

Reply:
@adamsilverstein I understand, my point was more there is another way to "fix" this which doesn't involve putting a non-image MIME type into the set of image transformations, which is a bit odd. WordPress core specifically passes the `full` JPEG image already generated to `_wp_make_subsizes()` so we should be able to rely on that instead of supporting `application/pdf`. Also see my original comment https://github.com/WordPress/wordpress-develop/pull/2393#discussion_r828487902 which touches on this.